### ISSUE TITLE
Review: MoreEnumerable.Interleave()

### ISSUE THREAD

**Issue by GoogleCodeExporter** [blyzer/morelinq#56]
```
Purpose of code changes on this branch:

Rev 150: Implementation of the Interleave operator which combines two or
more sequences by visiting each in turn, and returning the first element of
each, then the second, third, and so forth. By default, Interleave will
skip sequences that have been fully consumed and continue interleaving the
remaining sequences. An overload is available where the caller may specify
the strategy applied when interleaving sequences of unequal length. The
options are Pad, Skip, and Stop. Pad adds default(T) elements in place of
each consumed sequence. Skip skips consumed sequences. Stop terminates the
interleave operation. This operator behaves in a deferred and streaming manner.

When reviewing my code changes, please focus on:

* The public interface of the extension method.
* The effectiveness and clarity of the available XML comment documentation.
* How well this operator fits into the MoreLINQ ecosystem of
extension methods.
* The extent of coverage available from the corresponding unit tests.
* Whether edge cases are correctly identified and handled in the
implementation and tests.
* Whether resource disposal logic will always correctly dispose of
enumerators as they are consumed.

After the review, I'll merge this branch into:
/trunk

```

Original issue reported on code.google.com by `ambientl...@gmail.com` on 25 Jan 2010 at 12:57

**Comment by GoogleCodeExporter** [comment]
```
Now migrated to Hg and available in clone:
http://code.google.com/r/azizatif-morelinq-evenmore/
This clone is ready to be cloned :O) for further review. It also has the 
main/default branch already merged in to bring it up to date.
```

Original comment by `azizatif` on 25 May 2012 at 11:38

**Comment by GoogleCodeExporter** [comment]
Original comment by `azizatif` on 8 Jun 2012 at 1:30
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Following no response from ambientlion (who possibly/assuming lost interest 
given it has taken years to get this far), taking ownership of issue to 
consider as addition for MoreLINQ 2.0 milestone.
```

Original comment by `azizatif` on 12 Jun 2013 at 10:10
- Added labels: **Component-Logic**, **Milestone-Release2.0**, **Priority-High**, **Type-Enhancement**
- Removed labels: **Priority-Medium**, **Type-Review**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision ecbb044b3f86.
```

Original comment by `azizatif` on 14 Jun 2013 at 5:18
- Changed state: **Fixed**
